draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1062,98 +923,338 @@ trace has a different value for a given field, this field MUST NOT be added to
 common_fields but instead defined on each event individually. Good example of such
 fields are "time" and "data", who are divergent by nature.
 
-# Raw packet and frame information {#raw-info}
-
-While qlog is a high-level logging format, it also allows the inclusion of most
-raw wire image information, such as byte lengths and byte values. This is useful
-when for example investigating or tuning packetization behavior or determining
-encoding/framing overheads. However, these fields are not always necessary, can
-take up considerable space, and can have a considerable privacy and security
-impact (see {{privacy}}). Where applicable, these fields are grouped in a
-separate, optional, field named "raw" of type RawInfo. The exact definition of
-entities, headers, trailers and payloads depend on the protocol used.
-
-~~~ cddl
-RawInfo = {
-
-    ; the full byte length of the entity (e.g., packet or frame),
-    ; including possible headers and trailers
-    ? length: uint64
+# Concrete Event Types and Event Schema {#event-types-and-schema}
 
-    ; the byte length of the entity's payload,
-    ; excluding possible headers or trailers
-    ? payload_length: uint64
+Concrete event types are defined in event schema, which declare a namespace
+consisting of one or more categories, each containing related event types. This
+document defines the `main` event schema. Other examples are the `quic`
+{{QLOG-QUIC}} and `h3` {{QLOG-H3}} event schema.
 
-    ; the (potentially truncated) contents of the full entity,
-    ; including headers and possibly trailers
-    ? data: hexstring
-}
-~~~
-{: #raw-info-def title="RawInfo definition"}
+Concrete events MAY extend any part of the abstract Event class, including extending the "data" field ({{data-field}}) of adding custom fields.
 
-The RawInfo:data field can be truncated for privacy or security purposes, see
-{{truncated-values}}. In this case, the length and payload_length fields should
-still indicate the non-truncated lengths when used for debugging purposes.
+Event schema MUST define a registered non-empty namespace of type `text`.
 
-This document does not specify explicit header_length or trailer_length fields.
-In protocols without trailers, header_length can be calculated by subtracting
-the payload_length from the length. In protocols with trailers (e.g., QUIC's
-AEAD tag), event definition documents SHOULD define how to support header_length
-calculation.
+Event categories MUST belong to a single event schema. The MUST have a
+registered non-empty globally-unique identifier of type `text` and MUST have a
```

Comment:
This is not entirely how I understood things... categories do not have to be GLOBALLY unique, just within their specific namespace (e.g., you could have  namespaceX:categoryA and also namespaceY:categoryA). As such, the COMBINATION of namespace + category identifier should be globally unique. 

Additionally: I think it's better to say `Event categories MUST belong to a single event namespace` instead of to a single event `schema`... helps to reinforce the URI structure later.

Reply:
That creates a whole class of problem of ambiguity I'd like to avoid. If events dont self identify the namespace, and we dont make declaration of all schema a MUST, its a recipe for clashes where people add a super generic category like `util` and have an event name like `util:do_common_thing`. 

Instead if we operate the old school C methodology of embedding the unique token in the signature, we don't need to rely on namespace mangling. 

If we don't want to go this way, that's OK but we need a way to avoid name collisions when the event names could clash

Reply:
Discussed during editors' meeting.

Lucas is in the right here, should be ok as the current text states. Robin was conflating with something else.

Robin would like us to explicitly call out how things can be extended though:
- CANNOT add new events to existing categories in other documents
- CANNOT create new categories that have the same name as existing categories in other documents (this is basically what the current text above says)
- CAN create new categories in existing NAMESPACES as long as the category names are globally unique

And these should also be supported by a concrete example.